vllm/model_executor/models/phi3v.py

```diff
@@ -389,15 +390,61 @@ def _apply_hf_processor(
         processed_outputs['input_ids'] = token_ids
         return processed_outputs
 
-    def _get_dummy_mm_kwargs(
+    def _get_prompt_replacements(
+        self,
+        mm_items: MultiModalDataItems,
+        hf_inputs: BatchFeature,
+        mm_processor_kwargs: Mapping[str, object],
+    ) -> list[PromptReplacement]:
+        hf_processor = self._get_hf_processor()
+        image_tokens: list[str] = hf_processor.img_tokens  # type: ignore
+
+        mm_config = self.ctx.get_mm_config()
+        max_images = mm_config.limit_per_prompt.get("image", 1)
+
+        def get_replacement_phi3v(item_idx: int):
+            image_size = mm_items.get_image_size(item_idx)
+            num_tokens = get_phi3v_image_feature_size(
+                input_width=image_size.width,
+                input_height=image_size.height,
+                num_crops=mm_processor_kwargs.get("num_crops", 16),
+            )
```

Comment:
I think we can just reuse the `calc_num_image_tokens` or `calc_num_image_tokens_from_image_size` from processor instead of re-implement img tokens calculation: https://huggingface.co/microsoft/Phi-3.5-vision-instruct/blob/4a0d683eba9f1d0cbfb6151705d1ee73c25a80ca/processing_phi3_v.py#L380-L397